### ISSUE TITLE
HTML entities added to innerHTML are not decoded

### ISSUE THREAD

**Issue by westonruter** [ampproject/worker-dom#613]
Consider this `amp-script`:

```html
  <amp-script script="entity-test">
    <p>
      Loading...
    </p>
  </amp-script>
  <script id="entity-test" type="text/plain" target="amp-script">
  	document.querySelector('p').innerHTML = '&lsaquo;&#8249;&#x2039; &#x203A;&#8250;&rsaquo;';
  </script>

  <!-- Invalid AMP for debugging difference in innerHTML handling -->
  <button type=button onclick="eval( document.getElementById( 'entity-test' ).textContent )">
    Try running script in main thread
  </button>
```

The rendered output is:

```
&lsaquo;&#8249;&#x2039; &#x203A;&#8250;&rsaquo;
```

When I expected the HTML entities to be decoded as:

```html
‹‹‹ ›››
```

So WorkerDOM is not properly decoding HTML entities.

**Comment by kristoferbaxter** [comment]
Am investigating solutions here given there is no JS API I'm aware of that decodes html entities.

**Comment by jridgewell** [comment]
The label based ones need a map, the integers can call `String.fromCharCode`.

**Comment by westonruter** [comment]
Yes, I was just going to say: Decoding of numeric (decimal and hexadecimal) entities seems straightforward. Named entities, however, will need a lookup table.

**Comment by samouri** [comment]
The HTML spec website has a list of all the html entities: https://html.spec.whatwg.org/entities.json

**Comment by jridgewell** [comment]
Yup, but that table would be 10kb brotli'd, even after we remove the `codepoints` key (which we wouldn't need).

**Comment by samouri** [comment]
What do folks think about exposing a new function, `innerHTMLAsync`. It would directly send over the text to the main-thread to apply via `innerHTML`, and then worker-dom would refresh the vdom for the relevant node (delete/rehydrate)

**Comment by samouri** [comment]
I've created a PR that just decodes numeric entities (https://github.com/ampproject/worker-dom/pull/947). I think it could also be worth decoding the 4 named entities required to encode html, since they are by far the most common (< >, &, ").
